pkg/adapter/v2/cloudevents.go

```diff
@@ -99,28 +100,46 @@ func (c *client) reportCount(ctx context.Context, event cloudevents.Event, resul
 
 	if cloudevents.IsACK(result) {
 		var res *http.Result
+
 		if !cloudevents.ResultAs(result, &res) {
-			return fmt.Errorf("protocol.Result is not http.Result")
+			return c.reportError(reportArgs, result)
```

Comment:
I wonder if you should use a go channel to not block the main adapter thread.

Reply:
maybe. It's not a very expensive operation but worth exploring. Can we do this in a separate PR?

Reply:
/lgtm
/approve